### PR TITLE
[CARBONDATA-2704] Index file size in describe formatted command is not updated correctly with the segment file

### PR DESCRIPTION
**Pull request by dhatchayani**
**Problem:**
Describe formatted command is not showing correct index files size after index files merge.
**Solution:**
Segment file should be updated with the actual index files size of that segment after index files merge.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [x] Testing done
        UT added
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6939/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5722/

**Comment by dhatchayani** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6944/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5727/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5712/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala [hunk 1]
```diff
@@ -193,6 +196,30 @@ class CarbonIndexFileMergeTestCase
...
+        | CREATE TABLE fileSize(id INT, name STRING, city STRING, age INT)
+        | STORED BY 'org.apache.carbondata.format'
+        | TBLPROPERTIES('SORT_COLUMNS'='city,name')
+      """.stripMargin)
+    sql(s"LOAD DATA LOCAL INPATH '$file2' INTO TABLE fileSize OPTIONS('header'='false')")
+    val table = CarbonMetadata.getInstance().getCarbonTable("default", "fileSize")
+    var loadMetadataDetails = SegmentStatusManager
+      .readTableStatusFile(CarbonTablePath.getTableStatusFilePath(table.getTablePath))
+    var segment0 = loadMetadataDetails.filter(x=> x.getLoadName.equalsIgnoreCase("0"))
+    Assert.assertEquals(692, segment0.head.getIndexSize.toLong)
```
Comment (manishgupta88) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala]: Do not assert for hard coded values. Instead create the file path and take its size using CarbonFile getSize interface method

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala [hunk 2]
```diff
@@ -193,6 +196,30 @@ class CarbonIndexFileMergeTestCase
...
+    new CarbonIndexFileMergeWriter(table)
+      .mergeCarbonIndexFilesOfSegment("0", table.getTablePath, false)
+    loadMetadataDetails = SegmentStatusManager
+      .readTableStatusFile(CarbonTablePath.getTableStatusFilePath(table.getTablePath))
+    segment0 = loadMetadataDetails.filter(x=> x.getLoadName.equalsIgnoreCase("0"))
+    Assert.assertEquals(741, segment0.head.getIndexSize.toLong)
```
Comment (manishgupta88) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala]: same comment as above

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5816/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7096/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5872/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7178/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5954/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**